datafusion/expr/src/expr_schema.rs

```diff
@@ -322,10 +322,16 @@ impl ExprSchemable for Expr {
                 }
             }
             Expr::Cast(Cast { expr, .. }) => expr.nullable(input_schema),
-            Expr::AggregateFunction(AggregateFunction { func_def, .. }) => {
+            Expr::AggregateFunction(AggregateFunction { func_def, args, .. }) => {
                 match func_def {
+                    // TODO: min/max over non-nullable input should be recognized as non-nullable
                     AggregateFunctionDefinition::BuiltIn(fun) => fun.nullable(),
                     // TODO: UDF should be able to customize nullability
+                    AggregateFunctionDefinition::UDF(udf)
+                        if udf.name() == "count" && args.len() <= 1 =>
```

Comment:
I think `args.len() <= 1` is unnecessary, or am I overlooking something?🤔

Reply:
Instead of pattern matching all the possible function name, we can add another method in UDAFImpl, so we can customize the nullability for each function, as least the non-null property also applied to arra_agg, min, max, and probably more.

Reply:
I file #11274 for this

Reply:
> I think `args.len() <= 1` is unnecessary, or am I overlooking something?🤔

I don't know of `count(a,b)` function. If it is added in the future, would it be necessarily always producing non-null values?

> Instead of pattern matching all the possible function name [...]

absolutely agree, that's why i add this comment: https://github.com/apache/datafusion/pull/11256#discussion_r1666020300
thanks for filing the issue

Reply:
> I don't know of `count(a,b)` function. If it is added in the future, would it be necessarily always producing non-null values?

It looks like it's already supported. I think it is also non-nullable.
```
DataFusion CLI v39.0.0
> create table t(a int, b int) as values(NULL, NULL);
0 row(s) fetched.
Elapsed 0.007 seconds.

> select count(distinct a, b) from t;
+-------------------------+
| count(DISTINCT t.a,t.b) |
+-------------------------+
| 0                       |
+-------------------------+
```

Reply:
count for multiple args are not well-supported, maybe we should return error for this?

```
statement ok
create table t(a int, b int) as values (1, null), (null, 2), (3, null);

query I
select count(a, b) from t;
----
0
```

Reply:
@jayzhan211 I'll file a new issue about it.

Reply:
Filed #11303